openml/config.py

```diff
@@ -37,6 +37,33 @@
 # Number of retries if the connection breaks
 connection_n_retries = _defaults['connection_n_retries']
 
+# variables to keep track of last used settings for when `use_example_configuration` is activated
+_last_used_key = None
+_last_used_server = None
+
+
+def start_use_example_configuration():
+    global server
+    global apikey
+    global _last_used_server
+    global _last_used_key
```

Comment:
For some reason, I could not access the variables local to the module, so I have to work with global variables here as well :( 
Either that or introduce some class which saves the information in class attributes, (e.g. `openml.config.someclass.start_use_example_configuration()`, but that seemed uglier to me).